### PR TITLE
[Maintenance] Deprecate unused configuration nodes

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | -
| License         | MIT

From what I see these nodes were never in use.

### CODE REVIEW

**Review by diimpp — COMMENTED** [review]
Will this change affect existing bundles at 1.13/1.14 sylius?

image

What if some end-user made use of it?

**Comment by NoResponseMate** [comment]
@diimpp
Just rechecked.
There's no problem when it's missing in regards to Sylius, it's not in use per se, and if it's additionally added to Order configuration, for instance, it makes no difference since it's a new node with no connection to the Resource one.
There is an issue though, when someone sets the `options` in their yaml config; the container will fail to build since the node is no longer there.
No idea why someone would use this node since it's not consumed in any way, but either way just to be safe I'll deprecate it instead.

**Comment by loic425** [comment]
@NoResponseMate You missed this doc => https://github.com/Sylius/SyliusResourceBundle/blob/1.12/docs/reference.md

File: src/Bundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -63,7 +63,9 @@ private function addResourcesSection(ArrayNodeDefinition $node): void
                     ->arrayPrototype()
                         ->children()
                             ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->end()
-                            ->variableNode('options')->end()
+                            ->variableNode('options')
+                                ->setDeprecated('sylius/resource-bundle', '1.11', 'The "%node%" node at "%path%" is deprecated and will be removed in 2.0.')
```
Comment (GSadee) [inline src/Bundle/DependencyInjection/Configuration.php]:
```suggestion
                                ->setDeprecated('sylius/resource-bundle', '1.12', 'The "%node%" node at "%path%" is deprecated and will be removed in 2.0.')
```

File: src/Bundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -80,7 +82,9 @@ private function addResourcesSection(ArrayNodeDefinition $node): void
                             ->end()
                             ->arrayNode('translation')
                                 ->children()
-                                    ->variableNode('options')->end()
+                                    ->variableNode('options')
+                                        ->setDeprecated('sylius/resource-bundle', '1.11', 'The "%node%" node at "%path%" is deprecated and will be removed in 2.0.')
```
Comment (GSadee) [inline src/Bundle/DependencyInjection/Configuration.php]:
```suggestion
                                        ->setDeprecated('sylius/resource-bundle', '1.12', 'The "%node%" node at "%path%" is deprecated and will be removed in 2.0.')
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**